pkg/services/handler.go

```diff
@@ -86,32 +89,36 @@ func (h *Handler) deployvClusterRancherCluster(obj interface{}) error {
 		return nil
 	}
 
-	project, err := h.LocalUnstructuredClient.Get(h.Ctx, gvk.ProjectManagementCattle, ns.GetLabels()["field.cattle.io/projectId"], h.ClusterName)
-	if err != nil {
-		return fmt.Errorf("failed to get project for vCluster's namespace [%s]: %w", service.Namespace, err)
-	}
-
-	provisioningCluster, err := h.LocalUnstructuredClient.GetFirstWithLabel(h.Ctx, gvk.ClusterProvisioningCattle, "loft.sh/vcluster-service-uid", string(service.GetUID()))
-	if err != nil && !errors.IsNotFound(err) {
+	provisioningCluster, err := h.LocalUnstructuredClient.GetFirstWithLabel(h.Ctx, gvk.ClusterProvisioningCattle, constants.LabelVClusterServiceUID, string(service.GetUID()))
+	if client.IgnoreNotFound(err) != nil {
 		return fmt.Errorf("error getting provisioning cluster for vCluster instance: %w", err)
 	}
 
-	if errors.IsNotFound(err) {
+	if kerrors.IsNotFound(err) {
 		logger.Info("provisioning cluster does not exist for vCluster instance, creating...")
+		project, err := h.LocalUnstructuredClient.Get(h.Ctx, gvk.ClusterProvisioningCattle, ns.GetLabels()["field.cattle.io/projectId"], h.ClusterName)
```

Comment:
Check to see if project ID is empty. If it is then use the no project value. I don't think we need to get it in this case but if we do, then we should at least check that the error is a NotFound error.